src/js/components/Calendar/__tests__/Calendar-test.js

```diff
@@ -146,7 +161,288 @@ describe('Calendar', () => {
     );
     expect(container.firstChild).toMatchSnapshot();
     fireEvent.click(getByText('17'));
-    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2018-01-17T/));
+    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2020-01-17T/));
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('first day sunday week monday', () => {
+    // When the first day of the month is Sunday,
+    // and the request of firstDayOfWeek
+    // is Monday, we are verifing we are not missing a week, issue 3253.
+    const { container } = render(
+      <Grommet>
+        <Calendar
+          firstDayOfWeek={1}
+          date="2020-03-01T00:00:00-08:00"
+          animate={false}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('change months', () => {
+    jest.useFakeTimers();
+    const { container, getByLabelText } = render(
+      <Grommet>
+        <Calendar date={DATE} />
+      </Grommet>,
+    );
+    // Change the Calendar from January to December
+    fireEvent.click(getByLabelText('December 2019'));
+    act(() => {
+      jest.advanceTimersByTime(400);
+    });
     expect(container.firstChild).toMatchSnapshot();
+    // Change the Calendar back to January
+    fireEvent.click(getByLabelText('January 2020'));
+    act(() => {
+      jest.advanceTimersByTime(400);
+    });
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('select date with range', () => {
+    const onSelect = jest.fn();
+    const { getByText } = render(
+      <Grommet>
+        <Calendar date={DATE} onSelect={onSelect} range animate={false} />
+      </Grommet>,
+    );
+    fireEvent.click(getByText('11'));
+    expect(onSelect).toBeCalledWith([
+      [
+        expect.stringMatching(/2020-01-11T/),
+        expect.stringMatching(/^2020-01-15T/),
+      ],
+    ]);
+    fireEvent.click(getByText('20'));
+    expect(onSelect).toHaveBeenNthCalledWith(2, [
```

Comment:
Why do we need `NthCalled`?

Throughout this file, It seems like sometimes we us `toHaveBeenNthCalledWith` and other we use `toBeCalledWith`. It seems like we're only calling these after a specific event, which seems like we could always use `toBeCalledWith`.

Reply:
I changes the tests so that they all use `toBeCalledWith` instead of `toHaveBeenCalledWith` or `toHaveBeenNthCalledWith`